staging/src/k8s.io/apiserver/pkg/storage/storagebackend/config.go

```diff
@@ -126,3 +132,50 @@ func NewDefaultConfig(prefix string, codec runtime.Codec) *Config {
 		Transport:            TransportConfig{TracerProvider: oteltrace.NewNoopTracerProvider()},
 	}
 }
+
+func (t *TransportConfig) LoadTLSConfigurations() error {
+	if len(t.KeyFile) != 0 && len(t.KeyData) == 0 {
+		keyData, err := os.ReadFile(t.KeyFile)
+		if err != nil {
+			return err
+		}
+		t.KeyData = keyData
+	}
+	if len(t.CertFile) != 0 && len(t.CertData) == 0 {
+		certData, err := os.ReadFile(t.CertFile)
+		if err != nil {
+			return err
+		}
+		t.CertData = certData
+	}
+	if len(t.TrustedCAFile) != 0 && len(t.TrustedCAData) == 0 {
+		trustCAData, err := os.ReadFile(t.TrustedCAFile)
+		if err != nil {
+			return err
+		}
+		t.TrustedCAData = trustCAData
+	}
+	return nil
+}
+
+func (t *TransportConfig) ClientConfig() (*tls.Config, error) {
+	if err := t.LoadTLSConfigurations(); err != nil {
+		return nil, err
+	}
+
+	caCertPool := x509.NewCertPool()
+	caCertPool.AppendCertsFromPEM(t.TrustedCAData)
+	tlsConfig := &tls.Config{MinVersion: tls.VersionTLS12, RootCAs: caCertPool}
```

Comment:
Do we need to set the encryption suite when setting up tls1.2?

Reply:
it's just a copy from etcd config